From d1cfec7530bafd80d1a765d64e8fa9f36c520edd Mon Sep 17 00:00:00 2001
From: Alistair Delva <adelva@google.com>
Date: Mon, 31 Aug 2020 08:11:40 -0700
Subject: [PATCH 760/830] ANDROID: GKI: Disable FW_LOADER_USER_HELPER_FALLBACK
X-NVConfidentiality: public

As documented by the Kconfig help text, the right way to enable the
fallback (if needed) is to echo 1 into
/proc/sys/kernel/firmware_config/force_sysfs_fallback

Setting the fallback globally causes delays when using the in-kernel
firmware loader if the driver module is loaded in first stage init, even
if the driver would correctly retry direct firmware loading some time
later.

Any driver relying on the FW_LOADER_USER_HELPER should be migrated to
use the kernel's direct firmware loader so that init/ueventd does not
need to load the firmware for the driver, which is less secure.

Bug: 163060741
Change-Id: Ifc45ff42e846aed03263b4f1eb584cc09fe96933
Signed-off-by: Alistair Delva <adelva@google.com>
---
 arch/arm64/configs/gki_defconfig | 1 -
 arch/x86/configs/gki_defconfig   | 1 -
 2 files changed, 2 deletions(-)

diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
index 47696de7f33f..c3e101986b83 100644
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -217,7 +217,6 @@ CONFIG_PCI_HOST_GENERIC=y
 CONFIG_PCIE_QCOM=y
 CONFIG_PCIE_KIRIN=y
 CONFIG_FW_LOADER_USER_HELPER=y
-CONFIG_FW_LOADER_USER_HELPER_FALLBACK=y
 # CONFIG_FW_CACHE is not set
 # CONFIG_ALLOW_DEV_COREDUMP is not set
 # CONFIG_SUN50I_DE2_BUS is not set
diff --git a/arch/x86/configs/gki_defconfig b/arch/x86/configs/gki_defconfig
index c53381963ff0..cf21ed75bd0e 100644
--- a/arch/x86/configs/gki_defconfig
+++ b/arch/x86/configs/gki_defconfig
@@ -193,7 +193,6 @@ CONFIG_PCI=y
 CONFIG_PCIEPORTBUS=y
 CONFIG_PCI_MSI=y
 CONFIG_FW_LOADER_USER_HELPER=y
-CONFIG_FW_LOADER_USER_HELPER_FALLBACK=y
 # CONFIG_FW_CACHE is not set
 # CONFIG_ALLOW_DEV_COREDUMP is not set
 CONFIG_GNSS=y
-- 
2.17.1

